stackslib/src/chainstate/nakamoto/miner.rs

```diff
@@ -521,6 +531,37 @@ impl NakamotoBlockBuilder {
             .block_limit()
             .expect("Failed to obtain block limit from miner's block connection");
 
+        let mut soft_limit = None;
+        if let Some(percentage) = settings
+            .mempool_settings
+            .tenure_cost_limit_per_block_percentage
+        {
+            // Make sure we aren't actually going to multiply by 0 or attempt to increase the block limit.
+            assert!(
```

Comment:
This is also checked when the config is loaded / initialized, right?

Reply:
Yes, it is. See this earlier discussion - https://github.com/stacks-network/stacks-core/pull/5450#discussion_r1844369879